### PR TITLE
Listed Hashicorp Plugin Page 

### LINKED ISSUE #1587: List hashicorp plugin on plugins page
We should link to the @dev-drprasad 's hashicorp plugin on https://porter.sh/plugins, just like we do for the azure plugin.

* Here is where we need a page for hashicorp: https://github.com/getporter/porter/tree/main/docs/content/plugins
* The page should contain a subset of the information on the readme, the same type of information as with the azure plugin: https://github.com/dev-drprasad/porter-hashicorp-plugins

### PR DESCRIPTION
**Pull request by DARK-art108**
# What does this change
Added Docs for Integrating Porter with Hashicorp Vault.

# What issue does it fix
Closes #1587 

[1]: /CONTRIBUTING.md#when-to-open-a-pull-request

# Notes for the reviewer
_Put any questions or notes for the reviewer here._

# Checklist
- [ ] Unit Tests
- [x] Documentation
- [ ] Schema (porter.yaml)

If this is your first pull request, please add your name to the bottom of our [Contributors][contributors] list. Thank you for making Porter better! 🙇‍♀️

[contributors]: /CONTRIBUTORS.md

### CODE REVIEW

**Comment by DARK-art108** [comment]
@carolynvs Added Hashicorp Plugin Page Take a look at it! :)

File: docs/content/plugins/hashicorp.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
+title: Hashicorp plugin
+description: Integrate Porter with Hashicorp
+---
+
+<img src="/images/mixins/hashicorp.png" class="mixin-logo" style="width: 300px"/>
```
Comment (carolynvs) [inline docs/content/plugins/hashicorp.md:6]: The image should go in `/images/plugins` and this link updated with the new location.

**Review by carolynvs — COMMENTED** [review]
Looks good, just one small change and this is ready to merge.

**Comment by DARK-art108** [comment]
@carolynvs Added suggested Changes, there is no Plugins folder previously, I have added it and dropped the png into that and updated the path! Take a look at it :)

**Review by carolynvs — APPROVED** [review]
LGTM, thanks!

**Pull request merged**